internal/kubernetes/operator/config_exporter_test.go

```diff
@@ -416,3 +416,246 @@ func TestExportAtlasStreamProcessing(t *testing.T) {
+									},
+								},
+							},
+						},
+					},
+					Status: akov2status.AtlasFederatedAuthStatus{
+						Common: akoapi.Common{
+							Conditions: []akoapi.Condition{},
+						},
+					},
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "should return nothing because the IDP is not active",
+			setupMocks: func(store *mocks.MockOperatorGenericStore, featureValidator *mocks.MockFeatureValidator) {
+				federationSettings := &admin.OrgFederationSettings{
+					Id:                     pointer.Get("TestFederationSettingID"),
+					IdentityProviderStatus: pointer.Get("INACTIVE"),
+					IdentityProviderId:     pointer.Get(legacyTestIdentityProviderID),
+					HasRoleMappings:        pointer.Get(false),
+				}
+				featureValidator.EXPECT().
+					IsResourceSupported(features.ResourceAtlasFederatedAuth).
+					Return(true)
+				store.EXPECT().FederationSetting(&admin.GetFederationSettingsApiParams{OrgId: orgID}).
+					Return(federationSettings, nil)
+			},
+			expected:      nil,
+			expectedError: nil,
+		},
+		{
+			name: "should return nothing because the IDP is not present",
+			setupMocks: func(store *mocks.MockOperatorGenericStore, featureValidator *mocks.MockFeatureValidator) {
+				federationSettings := &admin.OrgFederationSettings{
+					Id:              pointer.Get("TestFederationSettingID"),
+					HasRoleMappings: pointer.Get(false),
+				}
+				featureValidator.EXPECT().
+					IsResourceSupported(features.ResourceAtlasFederatedAuth).
+					Return(true)
+				store.EXPECT().FederationSetting(&admin.GetFederationSettingsApiParams{OrgId: orgID}).
+					Return(federationSettings, nil)
+			},
+			expected:      nil,
+			expectedError: nil,
+		},
+		{
+			name: "should return nil when resource is not supported",
+			setupMocks: func(_ *mocks.MockOperatorGenericStore, featureValidator *mocks.MockFeatureValidator) {
+				featureValidator.EXPECT().
+					IsResourceSupported(features.ResourceAtlasFederatedAuth).
+					Return(false)
+			},
+			expected:      nil,
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctl := gomock.NewController(t)
+			atlasOperatorGenericStore := mocks.NewMockOperatorGenericStore(ctl)
+			featureValidator := mocks.NewMockFeatureValidator(ctl)
+			ce := defaultTestConfigExporter(t, atlasOperatorGenericStore, featureValidator)
+			defer ctl.Finish()
+			tc.setupMocks(atlasOperatorGenericStore, featureValidator)
+
+			resources, err := ce.exportAtlasFederatedAuth("my-project")
+			require.Equal(t, tc.expectedError, err)
+			assert.Equal(t, tc.expected, resources)
+		})
+	}
+}
+func defaultTestConfigExporter(t *testing.T, genStore *mocks.MockOperatorGenericStore, featureValidator *mocks.MockFeatureValidator) *ConfigExporter {
+	t.Helper()
+	return NewConfigExporter(genStore, nil, projectID, orgID).
+		WithTargetNamespace("test").
+		WithFeatureValidator(featureValidator).
+		WithTargetOperatorVersion("2.3.0").
```

Comment:
We probably need to bump now to `2.4.0` here:
```suggestion
		WithTargetOperatorVersion("2.4.0").
```